pkg/webhook/podspec_dryrun.go

```diff
@@ -45,7 +45,7 @@ func decodeTemplate(ctx context.Context, val interface{}) (*v1.RevisionTemplateS
 	return templ, nil
 }
 
-func validatePodSpec(ctx context.Context, ps v1.RevisionSpec, namespace string) *apis.FieldError {
+func validatePodSpec(ctx context.Context, ps v1.RevisionSpec, namespace, mode string) *apis.FieldError {
```

Comment:
this should probably be a `enum`, akin to https://github.com/knative/serving/blob/master/pkg/apis/serving/v1/revision_types.go#L92
Basically a type aliased to `string` with a   constant values.

Reply:
How about just some bare constants? Does the type help?

Reply:
Hmm the map to get from string to type is funny. Am I doing this right?